front/src/modules/ui/table/states/selectors/hiddenTableColumnsScopedSelector.ts

```diff
@@ -1,13 +1,22 @@
 import { selectorFamily } from 'recoil';
```

Comment:
same remark here actually?

Reply:
Ok, so to clarify:
- I think we only need one state: savedTableColumnsState
- then we can have two selectors:
-- savedTableColumnSelector ==> represents the columns that can be removed
-- (optional) hiddenColumnsSelector which is doing availableColumns (static) - savedTableColumnState ==> represents the columns that can be added

hiddenColumnsSelector is optional as it can be computed in the componenent / in a hook based on savedTableColumnSelector if we want.
But I like having a selector that does that.

Reply:
Does this mean we want to delete columns on "hide"? If so, then we won't be able to retrieve their custom size when they are re-added, the size will be reset to default (and later if we want to allow changing columns order, we won't be able to retrieve a custom order as well, it will be reset).

This might also imply that the `isVisible` prop in DB is not used as all columns saved in DB are meant to be visible.

Right now we don't delete columns on "hide", we change their visibility to `isVisible: false`.
Hidden columns either:
- exist in DB, but have `isVisible: false`
- do not exist in DB ("available").

With the current approach, we cannot compute hidden columns with `available (static) - visible` because we use the `isVisible: false` columns DB ids to determine if the column exists in DB and should be updated with `isVisible: true` or if it doesn't exist and should be created. `available (static)` does not contain those ids. So we need to compute hidden columns with `isVisible: false` columns + `available (static) - local columns`.

(By "local" columns I mean `tableColumnsScopedState`. `savedTableColumnsScopedState` contain the persisted columns and is used to compute a diff with the local `tableColumnsScopedState` and know which columns need to be persisted).

Reply:
Yes, I meant `tableColumnsScopedState` and not `tableColumnsScopedState`!

No, we don't want to delete columns on "hide" ;
Why: hidden columns = available (static) - tableColumnsScopedState.filter(visible = true) would not work?

Reply:
I tried `available (static) - tableColumnsScopedState.filter(visible = true)` at first, but I had a problem with duplicated columns being created.

When a column is added/removed, the `onAddColumn(column)` and `handleColumnVisibilityChange(column)` methods test if `column.id` exists in `columnsById` (which means: the column to add exists in DB, but its visibility needs to be updated).
If `column.id` does not exist in `columnsById`, it means the column does not exist in DB, it's an "available" column. Then we know we need to create this column in DB (with `isVisible: true`).

With `hidden columns = available (static) - tableColumnsScopedState.filter(visible = true)`, the `column.id` information from DB is lost for hidden columns (we only have the dummy default ids from `available (static)`), so testing if the id already existed always returned false, so we created a column every time, even though the DB column already existed... which caused an error because of the unique constraint on `ViewField`.

If we don't rely on the id, every time when we add/change column visibility, we would have to rely on `fieldName`, test every column in `tableColumnsScopedState` to see if the `fieldName` already exists and find the id anyway (because we need the id to update the column in DB).

So instead of having the id readily available: `columnsById[column.id] ? ... : ...`
We would have to find the column first : `const previousColumn = columns.find(({ fieldName }) => fieldName === column.fieldName)`
To test if it exists, get its id and update it.

Moreover, I'd find it confusing not to be able to rely on `column.id` (and other DB data) for columns in `hiddenTableColumnsScopedSelector` when columns in `visibleTableColumnsScopedSelector` have the correct DB data.